java/client/src/org/openqa/selenium/support/events/EventFiringWebDriver.java

```diff
@@ -307,43 +355,93 @@ public Mouse getMouse() {
 
   public TouchScreen getTouch() {
     if (driver instanceof HasTouchScreen) {
-      return new EventFiringTouch(driver, dispatcher);
+      return new EventFiringTouch(driver, castDispatcher(TouchEventLitener.class));
     } else {
       throw new UnsupportedOperationException("Underlying driver does not implement advanced"
           + " user interactions yet.");
     }
   }
+  
+  private class EventFiringAlert implements Alert{
+	private final Alert alert;
```

Comment:
It might help to run a java code formatter since this doesn't conform to the rest of the code style.